benchmarking/runs/priority_algorithm/internal_parameter_exploration/five_diversity_constraint.py

```diff
@@ -0,0 +1,228 @@
+from benchmarking.simulation.insight import Insight
+from benchmarking.simulation.simulation_set import SimulationSetArtifact, SimulationSet
+from benchmarking.simulation.simulation_settings import SimulationSettings
+
+
+class FiveDiversityConstraint(Run):
+    def start(self, num_trials: int = 30, generate_graphs: bool = True):
+        class_sizes = [120]
+        num_teams = 30
+        student_providers: Dict[int, StudentProvider] = {
+            120: CustomOneHundredAndTwentyStudentProvider(),
+        }
+
+        # Ranges
+        max_keep_range = [1] + list(range(5, 31, 5))
+        max_spread_range = [1] + list(range(5, 31, 5))
+        max_iterations_range = [1, 5, 10, 20, 30]
+
+        scenario = FiveDiversityConstraintScenario(
+            value_of_female=Gender.FEMALE.value,
+            value_of_a=Gpa.A.value,
+            value_of_21=21,
+            value_of_math=Major.MATH.value,
+            value_of_european=Race.European.value,
+        )
+        metrics = {
+            "PrioritySatisfaction": PrioritySatisfaction(
+                goals_to_priorities(scenario.goals),
+                False,
+            ),
+            "AverageCosineDifference": AverageCosineDifference(),
+            "AverageSoloStatus": AverageSoloStatus(
+                minority_groups={
+                    ScenarioAttribute.GENDER.value: [_.value for _ in Gender.values()],
+                    ScenarioAttribute.GPA.value: [_.value for _ in Gpa.values()],
+                    ScenarioAttribute.AGE.value: [_.value for _ in Age.values()],
+                    ScenarioAttribute.MAJOR.value: [_.value for _ in Major.values()],
+                    ScenarioAttribute.RACE.value: [_.value for _ in Race.values()],
+                }
+            ),
+        }
+        start_types = [
+            PriorityAlgorithmStartType.WEIGHT,
+            PriorityAlgorithmStartType.RANDOM,
+        ]
+
+        artifacts_dict = {
+            class_size: {start_type: {} for start_type in start_types}
+            for class_size in class_sizes
+        }
+        for class_size in class_sizes:
+            for start_type in start_types:
+                artifact: SimulationSetArtifact = SimulationSet(
+                    settings=SimulationSettings(
+                        num_teams=num_teams,
+                        scenario=scenario,
+                        student_provider=student_providers[class_size],
+                        cache_key=f"priority_algorithm/internal_parameter_exploration/class_size_{class_size}/five_diversity_constraint/25_buckets",
+                    ),
+                    algorithm_set={
+                        AlgorithmType.PRIORITY: [
+                            PriorityAlgorithmConfig(
+                                MAX_KEEP=max_keep,
+                                MAX_SPREAD=max_spread,
+                                MAX_ITERATE=max_iterations,
+                                MAX_TIME=1000001,
```

Comment:
nit;

why not a solid number


also MEGA nit;

you can write numbers with underscores to separate blocks of zeroes, for readability i.e. `1_000_000  # 1 million`

Reply:
haha cause this was written a while ago while we were brain dead, being silly, and just making stuff run. It didn't matter what the number was used here as long as it was far bigger than any reasonable runtime, but I can fix it now that's being merged